tests/apiserver-proxy-tests/fetch-token/internal/idtokenprovider.go

```diff
@@ -53,11 +53,14 @@ func (p *dexIdTokenProvider) implicitFlow() (map[string]string, error) {
 	if err != nil {
 		return nil, err
 	}
+	authorizeRespBody := readRespBody(authorizeResp)
+	defer closeRespBody(authorizeResp)
```

Comment:
Always close body right after checking the error. Not very critical in this case, but it is a good practice.

Reply:
@sjanota that was done to always read the body to completion before closing it - not only in the case of an error - as you suggested.

Reply:
You are right. We should defer closing the body before reading it.